### PR TITLE
Fixes for dolt installer bash-script

### PR DESCRIPTION
**Pull request by imcrazytwkr**
1. Fixed error message mistakenly implying that dolt can be installed on 32-bit systems
2. Fixed color printing on bash (and zsh running in bash compatibility mode) installed on an OS with non-GNU coreutils (Alpine Linux, macOS)
3. Linted `install.sh` and fixed all the errors and warning that had obvious fixes. The only one left is `local error_code="$1"` in `fail` function that should be removed but may be reused for debugging purposes by manually editing the script

### CODE REVIEW

**Comment by timsehn** [comment]
@reltuk should look at this...

File: go/utils/publishrelease/install.sh [hunk 1]
```diff
@@ -30,84 +30,88 @@ WORK_DIR=
...
 install_binary_release() {
-  local FILE=dolt-$PLATFORM_TUPLE.tar.gz
-  local URL=$RELEASES_BASE_URL/$FILE
-  echo "Downloading:" $URL
-  curl -A "$CURL_USER_AGENT" -fsL "$URL" > "$FILE"
+  local FILE="dolt-$PLATFORM_TUPLE.tar.gz"
+  local URL="$RELEASES_BASE_URL/$FILE"
+
+  echo "Downloading: $URL"
+  curl -a "$CURL_USER_AGENT" -fsL "$URL" > "$FILE"
```
Comment (reltuk) [inline go/utils/publishrelease/install.sh:107]: This change from `-A` to `-a` doesn't look right to me.

https://curl.se/docs/manpage.html#-A

**Review by reltuk — COMMENTED** [review]
Thanks for the contribution. This looks great.

One question about a change in the `curl` invocation.

File: go/utils/publishrelease/install.sh (hunk 1 above)
Comment (imcrazytwkr) [reply go/utils/publishrelease/install.sh:107]: Sorry, my bad on this one. Let me push a quick fix.

**Comment by reltuk** [comment]
@imcrazytwkr Thank you for the contribution! Awesome to have the script a bit better and cleaned up like this :).

**Pull request merged**